tdrs-backend/tdpservice/parsers/util.py

```diff
@@ -105,3 +259,59 @@ def transform_to_months(quarter):
             return ["Oct", "Nov", "Dec"]
         case _:
             raise ValueError("Invalid quarter value.")
+
+
+def month_to_int(month):
+    """Return the integer value of a month."""
+    return datetime.strptime(month, '%b').strftime('%m')
+
+
+def case_aggregates_by_month(df, dfs_status):
+    """Return case aggregates by month."""
+    section = str(df.section)  # section -> text
+    program_type = get_prog_from_section(section)  # section -> program_type -> text
+
+    # from datafile year/quarter, generate short month names for each month in quarter ala 'Jan', 'Feb', 'Mar'
+    calendar_year, calendar_qtr = fiscal_to_calendar(df.year, df.quarter)
+    month_list = transform_to_months(calendar_qtr)
+
+    short_section = get_text_from_df(df)['section']
+    schema_models_dict = get_program_models(program_type, short_section)
+    schema_models = [model for model in schema_models_dict.values()]
+
+    aggregate_data = {"months": [], "rejected": 0}
+    for month in month_list:
+        total = 0
+        cases_with_errors = 0
+        accepted = 0
+        month_int = month_to_int(month)
+        rpt_month_year = int(f"{calendar_year}{month_int}")
+
+        if dfs_status == "Rejected":
+            # we need to be careful here on examples of bad headers or empty files, since no month will be found
+            # but we can rely on the frontend submitted year-quarter to still generate the list of months
+            aggregate_data["months"].append({"accepted_with_errors": "N/A",
+                                             "accepted_without_errors": "N/A",
+                                             "month": month})
+            continue
+
+        case_numbers = set()
+        for schema_model in schema_models:
+            if isinstance(schema_model, SchemaManager):
+                schema_model = schema_model.schemas[0]
+
+            curr_case_numbers = set(schema_model.model.objects.filter(datafile=df).filter(RPT_MONTH_YEAR=rpt_month_year)
+                                    .distinct("CASE_NUMBER").values_list("CASE_NUMBER", flat=True))
+            case_numbers = case_numbers.union(curr_case_numbers)
+
+        total += len(case_numbers)
+        cases_with_errors += ParserError.objects.filter(case_number__in=case_numbers).distinct('case_number').count()
+        accepted = total - cases_with_errors
+
+        aggregate_data['months'].append({"month": month,
+                                         "accepted_without_errors": accepted,
+                                         "accepted_with_errors": cases_with_errors})
+
+    aggregate_data['rejected'] = ParserError.objects.filter(file=df).filter(case_number=None).count()
+
+    return aggregate_data
```

Comment:
i think there is another edge case: when the data file time period does not map to the frontend time period selections, the aggregates are all 0. the data are still being parsed, errors generated, but the aggregate values dont match whats in the feedback report (see below). should this still be parsed? i'd be inclined to reject and aggregates == 'n/a' 

![1613p6](https://github.com/raft-tech/TANF-app/assets/63075587/c948621d-743e-44ca-86d2-6ccd64b8f424)

Reply:
After discussing with the dev team, we have drafted #2711 to address this scenario in the future.